app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -299,13 +299,40 @@ def upload_content
 
     api :PUT, "/repositories/:id/import_uploads", N_("Import uploads into a repository")
     param :id, :identifier, :required => true, :desc => N_("Repository id")
-    param :upload_ids, Array, :required => true, :desc => N_("Array of upload ids to import")
+    param :upload_ids, Array, :desc => N_("Array of upload ids to import"), :deprecated => true
+    param :uploads, Array, :desc => N_("Array of uploads to import") do
+      param 'id', String, :required => true
+      param 'size', String
+      param 'checksum', String
+      param 'name', String
+    end
     def import_uploads
-      params[:upload_ids].each do |upload_id|
-        begin
-          sync_task(::Actions::Katello::Repository::ImportUpload, @repository, upload_id)
-        rescue => e
-          raise HttpErrors::BadRequest, e.message
+      if params['upload_ids'].empty? && params['uploads'].empty?
+        raise HttpErrors::BadRequest, e.message
```

Comment:
I think you may need to replace `e.message` with an actual message (e.g. `_("No upload params specified")`)